src/main/scala/modbat/mbt/ModelInstance.scala

```diff
@@ -112,13 +118,15 @@ class ModelInstance (val mbt: MBT, val model: Model,
   }
 
   def addAndLaunch(firstLaunch: Boolean) = {
+    var isFirstInstance = false
```

Comment:
Replace with `val isFirstInstance = !mbt.firstInstance.contains(className)` and then use that Boolean in the `if` statement below.